BUILD.bazel

```diff
@@ -739,15 +730,23 @@ ray_cc_binary(
 
 # Ray native pubsub module.
 ray_cc_library(
-    name = "pubsub_lib",
-    srcs = [
-        "src/ray/pubsub/publisher.cc",
-        "src/ray/pubsub/subscriber.cc",
-    ],
-    hdrs = [
-        "src/ray/pubsub/publisher.h",
-        "src/ray/pubsub/subscriber.h",
+    name = "publisher_lib",
+    srcs = ["src/ray/pubsub/publisher.cc"],
+    hdrs = ["src/ray/pubsub/publisher.h"],
+    deps = [
+        ":pubsub_rpc",
```

Comment:
I took a glance of code, it looks to me neither subscriber nor publisher uses `boost::any`, so we should be good to remove the header file and dependency.

One goal for the target splitting is to cleanup unnecessary dependency.